tests/ui/traits/bound/same-crate-name.stderr

```diff
@@ -12,6 +12,7 @@ help: trait impl with same name found
 LL | impl Bar for Foo {}
    | ^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `crate_a2` are being used?
+   = note: `Foo` implements similarly named `main::a::Bar`, but not `main::a::Bar`
```

Comment:
looking at this error message, we've already got another suggestion which is doing something similar. How do they differ? I feel like there should never be a case where we emit both of them, so probably their implementation should get merged into one